src/hal/src/lib.rs

```diff
@@ -375,6 +375,8 @@ pub trait Backend: 'static + Sized + Eq + Clone + Hash + fmt::Debug + Any + Send
     type Fence:               fmt::Debug + Any + Send + Sync;
     type Semaphore:           fmt::Debug + Any + Send + Sync;
     type QueryPool:           fmt::Debug + Any + Send + Sync;
+
+    type Window:                           Any + Send + Sync;
```

Comment:
This shouldn't be here, ideally.
Could you explain the thought process that leaded you to this decision? GL should not affect our HAL API.